### DISCUSSION TITLE
Changing configuration at runtime

### DISCUSSION THREAD

**Discussion by vplauzon** [dotnet/yarp#389] [General] [answered]
Hi,

I managed to get a POC running.

For the real thing, I would need to change the destination address of the clusters at runtime.

Basically, I need to use an environment variable to set the destination.  You can imagine that in Azure and depending on the deployment, my backend have a different host name.

Is there a way to override the configuration within the bootstrapping of ASP.NET without altering the appsettings.json?

Cheers,

VP

**Comment by Tratcher** [comment, marked as answer]
The Microsoft.Extensions IConfiguration system used by AspNetCore to read appsettings.json allows for layering config from different sources including files and environment variables. It should just be a matter of figuring out the right environment variable name to match the appsetings.json structure for the destination entry you need to add/update.
https://docs.microsoft.com/en-us/aspnet/core/fundamentals/configuration/?view=aspnetcore-3.1#environment-variables

The alternative is to use a full code based provider:
https://microsoft.github.io/reverse-proxy/articles/configproviders.html

**Reply by vplauzon** [reply]
That works very well!  Thank you.

**Reply by Tratcher** [reply]
Can you share the environment variable you worked out? I know they can be verbose.

**Reply by vplauzon** [reply]
$"ReverseProxy__Clusters__{cluster}Cluster__Destinations__{cluster}Cluster/destination1__Address"

where {cluster} resolves into the name I gave to the cluster in the config file.
